lib/fasta_generator.py

```diff
@@ -50,7 +50,9 @@ def determine_flanking_sequence_length(self, full_wildtype_sequence_length, pept
 
     def get_wildtype_subsequence(self, position, full_wildtype_sequence, wildtype_amino_acid_length, peptide_sequence_length, line):
         one_flanking_sequence_length = self.determine_flanking_sequence_length(len(full_wildtype_sequence), peptide_sequence_length, line)
-        peptide_sequence_length = 2 * one_flanking_sequence_length + wildtype_amino_acid_length
+        ##clip by wt sequence length, otherwise with deletions peptide_sequence_length may exceeds full wt sequence length,
+        ##and the code below tries extracting ranges beyond the wt sequence
+        peptide_sequence_length = min(2 * one_flanking_sequence_length + wildtype_amino_acid_length,len(full_wildtype_sequence))
```

Comment:
Can you please provide some test data for this change?

Reply:
I have added a test to the test harness. It runs with my change, and fails without it. The target length of 600 in the test case is contrived to exceed ORF length in your test `input.vcf`, but it my real application it was failing with target length 60
```
pvacseq generate_protein_fasta tests/test_data/generate_protein_fasta/input.vcf 600 tmp.fasta
```